File: backend/src/main/java/com/example/upload/domain/member/member/dto/MemberDto.java
package com.example.upload.domain.member.member.dto;

import com.example.upload.domain.member.member.entity.Member;
import lombok.Getter;
import org.springframework.lang.NonNull;

@Getter
public class MemberDto {
    @NonNull
    private long id;
    @NonNull
    private String nickname;
    @NonNull
    private String profileImgUrl;

    public MemberDto(Member member) {
        this.id = member.getId();
        this.nickname = member.getNickname();
        this.profileImgUrl = member.getProfileImgUrlOrDefault();
    }
}

File: backend/src/main/java/com/example/upload/domain/member/member/dto/MemberDto.kt
package com.example.upload.domain.member.member.dto

import com.example.upload.domain.member.member.entity.Member
import com.example.upload.domain.post.post.entity.Post
import lombok.Getter
import org.springframework.lang.NonNull

class MemberDto(
    val id:Long,
    val nickname:String,
    val profileImgUrl:String
) {

    constructor(member: Member) : this(
        id = member.id!!,
        nickname=member.nickname,
        profileImgUrl=member.profileImgUrlOrDefault
    )
}
